modules/lambda-in-vpc/main.tf

```diff
@@ -0,0 +1,119 @@
+# IAM
+# Base AssumeRole policy for Lambda execution.
+data "aws_iam_policy_document" "execution_lambda_policy" {
+  statement {
+    actions = [
+      "sts:AssumeRole",
+    ]
+
+    principals {
+      type = "Service"
+
+      identifiers = [
+        "lambda.amazonaws.com",
+        "apigateway.amazonaws.com",
+      ]
+    }
+  }
+}
+
+# Base policy for Lambda to execute.
+data "aws_iam_policy_document" "base_lambda_policy" {
+  statement {
+    actions = [
+      "logs:*",
+      "lambda:InvokeFunction",
+      "xray:PutTraceSegments",
+      "xray:PutTelemetryRecords",
+    ]
```

Comment:
Since this module is only for Lambdas in VPCs, would it make sense to add the VPC IAM permissions here?

```
"ec2:CreateNetworkInterface"
"ec2:DescribeNetworkInterfaces"
"ec2:DeleteNetworkInterface"
```

Reply:
Successfully deployed the credential-service with the module as is, without the additional permissions.  VPC connection was created.

```module.credential_service_lambda.aws_lambda_function.lambda: Creating...
  arn:                                                "" => "<computed>"
  description:                                        "" => " (stage: josh)"
  environment.#:                                      "" => "1"
  environment.0.variables.%:                          "" => "4"
  environment.0.variables.CONFIG_SOURCE:              "" => "parameter_store"
  environment.0.variables.LOG_LEVEL:                  "" => "DEBUG"
  environment.0.variables.PARAMETER_STORE_AWS_REGION: "" => "us-east-1"
  environment.0.variables.PARAMETER_STORE_PREFIX:     "" => "/services/credential-service/josh"
  filename:                                           "" => "./dist/lambda_function.zip"
  function_name:                                      "" => "josh_credential-service"
  handler:                                            "" => "handler.lambda_handler"
  invoke_arn:                                         "" => "<computed>"
  last_modified:                                      "" => "<computed>"
  memory_size:                                        "" => "256"
  publish:                                            "" => "true"
  qualified_arn:                                      "" => "<computed>"
  role:                                               "" => "arn:aws:iam::013190989741:role/josh_credential-service_lambda"
  runtime:                                            "" => "python3.6"
  source_code_hash:                                   "" => "XYg0jaybPZDLa5PWCETJxLd52BlGTpTPjEgbJX9G/Z0="
  timeout:                                            "" => "60"
  tracing_config.#:                                   "" => "<computed>"
  version:                                            "" => "<computed>"
  vpc_config.#:                                       "" => "1"
  vpc_config.0.security_group_ids.#:                  "" => "1"
  vpc_config.0.security_group_ids.3518507390:         "" => "sg-a4fb43d7"
  vpc_config.0.subnet_ids.#:                          "" => "1"
  vpc_config.0.subnet_ids.785998683:                  "" => "subnet-ce0736e2"
  vpc_config.0.vpc_id:                                "" => "<computed>"```